src/Utils.php

```diff
@@ -307,9 +307,10 @@ public static function streamFor($resource = '', array $options = [])
 
         switch (gettype($resource)) {
             case 'resource':
-                if (\stream_get_meta_data($resource)['uri'] === 'php://input') {
+                if (in_array(\stream_get_meta_data($resource)['uri'], ['php://input', 'php://memory'])) {
```

Comment:
Would it be different if we added a functional test?

Reply:
Yes, functional test could help to avoid this change.

But what do you mean by functional tests in this context?

Reply:
I know that we dont do any of these tests in this repo. 

If we spin up a process that acts as a server, then do real HTTP request to that server process and validate the output. Much like in Guzzle/guzzle.